moo.js

```diff
@@ -210,15 +228,20 @@
 
       groups.push(options)
 
+      match.forEach(function (pattern) {
+        unicodeState.transition(pattern)
+      })
+
       // convert to RegExp
       var pat = reUnion(match.map(regexpOrLiteral))
 
       // validate
-      var regexp = new RegExp(pat)
+      var flags = unicodeState.hasFlag() ? 'u' : ''
+      var regexp = new RegExp(pat, flags)
```

Comment:
Every time we build a RegExp we need to add the `u` flag, if applicable; otherwise it complains about "nothing to repeat" in patterns like `/\u{0075}+/`